packages/fxa-auth-server/lib/l10n/index.ts

```diff
@@ -0,0 +1,128 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { DOMLocalization, Localization } from '@fluent/dom';
+import { FluentBundle, FluentResource } from '@fluent/bundle';
+import { negotiateLanguages } from '@fluent/langneg';
+import { LocalizerBindings } from './bindings';
+import availableLocales from 'fxa-shared/l10n/supportedLanguages.json';
+import { EN_GB_LOCALES } from 'fxa-shared/l10n/otherLanguages';
+import { RendererBindings } from '../senders/renderer/bindings';
+
+type Bindings = LocalizerBindings | RendererBindings;
+export interface FtlIdMsg {
+  id: string;
+  message: string;
+}
+
+class Localizer {
+  protected readonly bindings: Bindings;
+
+  constructor(bindings: Bindings) {
+    this.bindings = bindings;
+  }
+
+  protected async fetchMessages(currentLocales: string[]) {
+    const fetched: Record<string, string> = {};
+    for (const locale of currentLocales) {
+      fetched[locale] = await this.fetchLocalizationMessages(locale);
```

Comment:
I noticed while working on #11862 that I had to [increase the timeout for some tests](https://github.com/mozilla/fxa/pull/11862/files#diff-2c8309e103dbe24426b7c6205d95553b7809b4a8497ee373b14183ebd9c968b5R27) and that may be happening again but for content-server tests, according to CI. I'm probably going to look at some of the synchronous functions / stuff we're `await`ing and see what I can parallelize with a `Promise.all`.

Reply:
Since some of the calls _do_ need to be synchronous, instead of creating an async version of `fetchResource`, I just [switched to using a `Promise.allSettled` here](https://github.com/mozilla/fxa/pull/11962/commits/6e179807518dc3bd7a2cac6602cbfbef1ef77d9b#diff-c90899b96410aab8b55066b50be869d1c0a11abb183f242b00df92c54612ebfaR36) but maybe it'd be cleaner to switch `fetchResource` to be async... I think we'd still need something to ensure everything resolved though, so 🤔

Reply:
Ah yes, there was a down stream dependency that had to be synchronous.... what was that again? I remember digging way into this and realize we'd have to get async support in a library to make this happen. Was it mjml?

By the way, my point about readFileSync is this. If you are calling readFileSync in a loop, it's my understanding wrapping a promise around it won't really give you any advantage. The synchronous read will block until its done. Node isn't multithreaded, so it's not like the reads happen in parallel just because you wrap something in a promise, however, if readFile (the async version) is used, at least the scheduler can context switch between the blocking i/o operations and you might see some performance gains, and at the very least something else in the even loop can be serviced.